pkg/controller/v1alpha1/localmodelnode/controller.go

```diff
@@ -133,13 +165,26 @@ func (c *LocalModelNodeReconciler) launchJob(ctx context.Context, jobName string
 	if err != nil {
 		return nil, err
 	}
-	if job != nil && reflect.DeepEqual(job.Spec.Template.Spec, dryrunJob.Spec.Template.Spec) {
+
+	// If we do not recreate job on purpose, return the existing job if it is the same as the expected job
+	if !recreateJob && job != nil && reflect.DeepEqual(job.Spec.Template.Spec, dryrunJob.Spec.Template.Spec) {
 		return job, nil
 	}
+
+	// Remove the existing job because either of them is true:
+	// 1. it is different from the expected job
+	// 2. recreate the job on purpose by setting recreateJob to true
 	if jobFound {
-		bg := metav1.DeletePropagationBackground
+		// If the job is still running, do not delete it
+		if recreateJob && job.Status.Succeeded == 0 {
+			c.Log.Info("New job is not successful yet, skipping deletion", "name", job.Name)
+			return job, nil
+		}
+		c.Log.Info("Deleting job.", "name", job.Name)
+		policy := metav1.DeletePropagationForeground
 		err = jobs.Delete(ctx, job.Name, metav1.DeleteOptions{
```

Comment:
One small issue - in my tests the delete request is successful, but it cannot create a new job immediately because the deletion is still in progress. It would work in the next reconciliation loop.

Reply:
Alternatively we could use generate name and use labels for filtering.

Reply:
IMO creating the job in the next reconciliation loop is fine since this shouldn't happen a lot.